controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -645,4 +645,39 @@ public static boolean isRerunOf(final List<Integer> segmentsToSeal,
 
         return historyRecordOpt;
     }
+
+    public static boolean isScaleInputValid(final List<Integer> segmentsToSeal,
+                                            final List<AbstractMap.SimpleEntry<Double, Double>> newRanges,
+                                            final byte[] segmentTable) {
+
+        List<AbstractMap.SimpleEntry<Double, Double>> oldRanges = segmentsToSeal.stream()
+                .map(segment -> SegmentRecord.readRecord(segmentTable, segment)
+                        .map(x -> new AbstractMap.SimpleEntry<>(x.getRoutingKeyStart(), x.getRoutingKeyEnd())))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+
+        return reduce(oldRanges).equals(reduce(newRanges));
```

Comment:
list.equals compares two lists by calling equals on each element at the same index in either of the list. 
list.equals javadoc: 
`Compares the specified object with this list for equality. Returns true if and only if the specified object is also a list, both lists have the same size, and all corresponding pairs of elements in the two lists are equal. (Two elements e1 and e2 are equal if (e1==null ? e2==null : e1.equals(e2)).) In other words, two lists are defined to be equal if they contain the same elements in the same order. This definition ensures that the equals method works properly across different implementations of the List interface.`

The elements we have are SimpleEntry<Double, Double> which is also matched for equality in the same way -- call equals on key and equals value. Double equals returns true if the double value is same.